features/step_definitions/account_creation.rb

```diff
@@ -10,54 +10,61 @@
   expect(page).to have_content "Log in"
 end
 
-When(/^I click "(.*?)"$/) do |arg1|
-  pending # express the regexp above with the code you wish you had
+When(/^I click the "(.*?)" link$/) do |arg1|
+  click_link arg1 
 end
 
-When(/^I click the button "(.*?)"$/) do |arg1|
+When(/^I click the "(.*?)" button$/) do |arg1|
   click_button arg1
 end
 
 When(/^I enter my email address$/) do
-  pending # express the regexp above with the code you wish you had
+ fill_in :user_email, with: 'test@example.com' 
 end
 
 When(/^I enter a password with correct confirmation$/) do
-  pending # express the regexp above with the code you wish you had
+ fill_in :user_password, with: "password" 
+ fill_in :user_password_confirmation, with: "password"  
 end
 
 Then(/^I am told to check my email for a confirmation link$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content "A message with a confirmation link has been sent to your email address. Please follow the link to activate your account" 
 end
 
 Then(/^I am sent a confirmation email$/) do
-  pending # express the regexp above with the code you wish you had
+  email = ActionMailer::Base.deliveries.first
+  expect(email.from).to eq(["no-reply@smashingboxes.com"])
+  expect(email.to).to eq(["test@example.com"])
+  expect(email.body).to have_content("confirm your account") 
 end
 
 When(/^I visit the link in that email$/) do
-  pending # express the regexp above with the code you wish you had
+  open_last_email_for("test@example.com")
+  visit_in_email("Confirm my account") 
 end
 
 Then(/^My email address becomes confirmed$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content "Your email address has been successfully confirmed"
 end
 
 When(/^I enter a password with incorrect confirmation$/) do
-  pending # express the regexp above with the code you wish you had
+  fill_in :user_password, with: "password"
+  fill_in :user_password_confirmation, with: "passowrd" 
 end
 
 Then(/^I am notified that my password confirmation does not match$/) do
-  pending # express the regexp above with the code you wish you had
+  expect(page).to have_content "Sign up 1 error prohibited this user from being saved: Password confirmation doesn't match"
```

Comment:
I'd be less specific here, and do `expect(page).to have_content "Password confirmation doesn't match"`, because you don't know how many errors you might have.